src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateRequestService.swift

```diff
@@ -0,0 +1,378 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+import HealthCertificateToolkit
+
+class HealthCertificateRequestService {
+
+	// MARK: - Init
+
+	init(
+		store: HealthCertificateStoring,
+		client: Client,
+		appConfiguration: AppConfigurationProviding,
+		digitalCovidCertificateAccess: DigitalCovidCertificateAccessProtocol = DigitalCovidCertificateAccess(),
+		healthCertificateService: HealthCertificateService
+	) {
+		#if DEBUG
+		if isUITesting {
+			let store = MockTestStore()
+
+			self.store = store
+			self.client = ClientMock()
+			self.appConfiguration = CachedAppConfigurationMock(store: store)
+			self.digitalCovidCertificateAccess = digitalCovidCertificateAccess
+			self.healthCertificateService = healthCertificateService
+
+			setup()
+
+			return
+		}
+		#endif
+
+		self.store = store
+		self.client = client
+		self.appConfiguration = appConfiguration
+		self.digitalCovidCertificateAccess = digitalCovidCertificateAccess
+		self.healthCertificateService = healthCertificateService
+
+		setup()
+	}
+
+	// MARK: - Internal
+
+	@DidSetPublished private(set) var testCertificateRequests = [TestCertificateRequest]() {
+		didSet {
+			Log.debug("Did set testCertificateRequests.")
+
+			if initialTestCertificateRequestsReadFromStore {
+				store.testCertificateRequests = testCertificateRequests
+			}
+
+			updateTestCertificateRequestSubscriptions(for: testCertificateRequests)
+		}
+	}
+
+	var didRegisterTestCertificate: ((String, TestCertificateRequest) -> Void)?
+
+	func registerAndExecuteTestCertificateRequest(
+		coronaTestType: CoronaTestType,
+		registrationToken: String,
+		registrationDate: Date,
+		retryExecutionIfCertificateIsPending: Bool,
+		labId: String?,
+		completion: ((Result<Void, HealthCertificateServiceError.TestCertificateRequestError>) -> Void)? = nil
+	) {
+		Log.info("[HealthCertificateService] Registering test certificate request: (coronaTestType: \(coronaTestType), registrationToken: \(private: registrationToken), registrationDate: \(registrationDate), retryExecutionIfCertificateIsPending: \(retryExecutionIfCertificateIsPending)", log: .api)
+
+		let testCertificateRequest = TestCertificateRequest(
+			coronaTestType: coronaTestType,
+			registrationToken: registrationToken,
+			registrationDate: registrationDate,
+			labId: labId
+		)
+
+		testCertificateRequests.append(testCertificateRequest)
+
+		executeTestCertificateRequest(
+			testCertificateRequest,
+			retryIfCertificateIsPending: retryExecutionIfCertificateIsPending,
+			completion: completion
+		)
+	}
+
+	// swiftlint:disable:next cyclomatic_complexity
+	func executeTestCertificateRequest(
```

Comment:
Can this be further broken down since we are refactoring it?

Reply:
I managed to extract a part into a separate func now :)